### PR TITLE
Add engine flag constants

### PR DESCRIPTION
**Pull request by vulcandth**
commit `Define & Port pret engin_flag constants` does not change the sha1. The second commit does due to removing a incorrect ENGINE flag.

I ended not deleting the engine deleting the engine flags as after the cleanup there was very little reason to. They can easily be reused, and I think deleting thing would needlessly shift all of the `ENGINE_*` constant values for little benefit. If you still want it done, I can do so.

### CODE REVIEW

File: constants/engine_flags.asm [hunk 1]
```diff
@@ -108,7 +105,7 @@
 	const ENGINE_DAILY_BUG_CONTEST
 	const ENGINE_FISH_SWARM
 	const ENGINE_VALERIE_MORNING_WALK
-	const ENGINE_DAILY_UNUSED_1
+	const_skip
```
Comment (Rangi42) [inline constants/engine_flags.asm:108]:
```suggestion
	const_skip ; unused
```

File: constants/wram_constants.asm [hunk 2]
```diff
@@ -390,3 +395,100 @@ DEF DAYCARELADY_ACTIVE_F         EQU 7
...
 	const THORTON_JOURNAL ; Battle Factory
 DEF NUM_POKEMON_JOURNALS EQU const_value
+
+; wUnlockedUnowns ::
+	const_def
+	const UNLOCKED_UNOWNS_A_TO_J_F
+	const UNLOCKED_UNOWNS_K_TO_Q_F
+	const UNLOCKED_UNOWNS_R_TO_W_F
+	const UNLOCKED_UNOWNS_X_TO_QUESTION_F
+	DEF NUM_UNOWN_SETS EQU const_value
```
Comment (Rangi42) [inline constants/wram_constants.asm:405]:
```suggestion
DEF NUM_UNLOCKED_UNOWN_SETS EQU const_value
```

File: constants/wram_constants.asm [hunk 3]
```diff
@@ -390,3 +395,100 @@ DEF DAYCARELADY_ACTIVE_F         EQU 7
...
+; wLuckyNumberShowFlag::
+DEF LUCKYNUMBERSHOW_GAME_OVER_F EQU 0
+
+; wDailyFlags
+	const_def
+	const DAILYFLAGS_KURT_MAKING_BALLS_F                     ; 0
+	const DAILYFLAGS_DAILY_BUG_CONTEST_F                     ; 1
+	const DAILYFLAGS_FISH_SWARM_F                            ; 2
+	const DAILYFLAGS_VALERIE_MORNING_WALK_F                  ; 3
+	const_skip
```
Comment (Rangi42) [inline constants/wram_constants.asm:416]:
```suggestion
	const_skip ; unused
```

File: constants/wram_constants.asm [hunk 4]
```diff
@@ -390,3 +395,100 @@ DEF DAYCARELADY_ACTIVE_F         EQU 7
...
+	const UNLOCKED_UNOWNS_A_TO_J_F
+	const UNLOCKED_UNOWNS_K_TO_Q_F
+	const UNLOCKED_UNOWNS_R_TO_W_F
+	const UNLOCKED_UNOWNS_X_TO_QUESTION_F
+	DEF NUM_UNOWN_SETS EQU const_value
+
+; wLuckyNumberShowFlag::
+DEF LUCKYNUMBERSHOW_GAME_OVER_F EQU 0
+
+; wDailyFlags
```
Comment (Rangi42) [inline constants/wram_constants.asm:410]:
```suggestion
; wDailyFlags::
```

File: constants/wram_constants.asm [hunk 5]
```diff
@@ -390,3 +395,100 @@ DEF DAYCARELADY_ACTIVE_F         EQU 7
...
+	const DAILYFLAGS_KURT_MAKING_BALLS_F                     ; 0
+	const DAILYFLAGS_DAILY_BUG_CONTEST_F                     ; 1
+	const DAILYFLAGS_FISH_SWARM_F                            ; 2
+	const DAILYFLAGS_VALERIE_MORNING_WALK_F                  ; 3
+	const_skip
+	const DAILYFLAGS_GOT_SHUCKIE_TODAY_F                     ; 5
+	const DAILYFLAGS_GOLDENROD_UNDERGROUND_MERCHANT_CLOSED_F ; 6
+	const DAILYFLAGS_FOUGHT_IN_TRAINER_HALL_TODAY_F          ; 7
+
+; wDailyFlags2
```
Comment (Rangi42) [inline constants/wram_constants.asm:421]:
```suggestion
; wDailyFlags2::
```

File: constants/wram_constants.asm [hunk 6]
```diff
@@ -390,3 +395,100 @@ DEF DAYCARELADY_ACTIVE_F         EQU 7
...
+	const DAILYFLAGS2_GYM_LEADER_REMATCH_1_F ; 0
+	const DAILYFLAGS2_GYM_LEADER_REMATCH_2_F ; 1
+	const DAILYFLAGS2_GYM_LEADER_REMATCH_3_F ; 2
+	const DAILYFLAGS2_DAILY_PHOTOGRAPH_F     ; 3
+	const DAILYFLAGS2_DAILY_SHRINE_VISIT_F   ; 4
+	const DAILYFLAGS2_DAILY_MYSTERY_GIFT_F   ; 5
+	const DAILYFLAGS2_RED_IN_MOUNT_SILVER_F  ; 6
+	const DAILYFLAGS2_LEAF_IN_NAVEL_ROCK_F   ; 7
+
+; wDailyFlags3
```
Comment (Rangi42) [inline constants/wram_constants.asm:432]:
```suggestion
; wDailyFlags3::
```

File: constants/wram_constants.asm [hunk 7]
```diff
@@ -390,3 +395,100 @@ DEF DAYCARELADY_ACTIVE_F         EQU 7
...
+	const DAILYFLAGS3_BOUGHT_LEFTOVERS_F  ; 0
+	const DAILYFLAGS3_TOOK_HYPER_TEST_F   ; 1
+	const DAILYFLAGS3_CHERYL_DONE_TODAY_F ; 2
+	const DAILYFLAGS3_RILEY_DONE_TODAY_F  ; 3
+	const DAILYFLAGS3_BUCK_DONE_TODAY_F   ; 4
+	const DAILYFLAGS3_MARLEY_DONE_TODAY_F ; 5
+	const DAILYFLAGS3_MIRA_DONE_TODAY_F   ; 6
+	const DAILYFLAGS3_ANABEL_DONE_TODAY_F ; 7
+
+; wDailyFlags4
```
Comment (Rangi42) [inline constants/wram_constants.asm:443]:
```suggestion
; wDailyFlags4::
```

File: constants/wram_constants.asm [hunk 8]
```diff
@@ -390,3 +395,100 @@ DEF DAYCARELADY_ACTIVE_F         EQU 7
...
+	const_def
+	const DAILYFLAGS4_SEASHORE_SHELL_BELL_F           ; 0
+	const DAILYFLAGS4_SHAMOUTI_RESTAURANT_CHALLENGE_F ; 1
+	const DAILYFLAGS4_FOUGHT_PSYCHIC_INVER_F          ; 2
+	const DAILYFLAGS4_BOUGHT_MOOMOO_MILK_F            ; 3
+	const DAILYFLAGS4_CHANGED_TRENDY_PHRASE_F         ; 4
+	const DAILYFLAGS4_ALL_HIDDEN_GROTTOES_F           ; 5
+
+; wWeeklyFlags
```
Comment (Rangi42) [inline constants/wram_constants.asm:452]:
```suggestion
; wWeeklyFlags::
```

File: constants/wram_constants.asm [hunk 9]
```diff
@@ -390,3 +395,100 @@ DEF DAYCARELADY_ACTIVE_F         EQU 7
...
+	const PLAYER_CAUGHT_MOLTRES_F  ; 7
+
+; wPlayerCaught2::
+	const_def
+	const PLAYER_CAUGHT_MEW_F       ; 0
+	const PLAYER_CAUGHT_MEWTWO_F    ; 1
+	const PLAYER_CAUGHT_CELEBI_F    ; 2
+	const PLAYER_CAUGHT_SUDOWOODO_F ; 3
+
+; wPlayerCaught Masks
```
Comment (Rangi42) [inline constants/wram_constants.asm:491]:
```suggestion
```

File: data/events/engine_flags.asm [hunk 10]
```diff
@@ -71,105 +69,104 @@ EngineFlags:
...
 
-	engine_flag wDailyFlags, 0 ; kurt making balls
-	engine_flag wDailyFlags, 1 ; daily bug contest?
-	engine_flag wDailyFlags, 2 ; special wilddata?
-	engine_flag wDailyFlags, 3 ; valeries morning walk
+	engine_flag wDailyFlags, DAILYFLAGS_KURT_MAKING_BALLS_F
+	engine_flag wDailyFlags, DAILYFLAGS_DAILY_BUG_CONTEST_F
+	engine_flag wDailyFlags, DAILYFLAGS_FISH_SWARM_F
+	engine_flag wDailyFlags, DAILYFLAGS_VALERIE_MORNING_WALK_F
 	engine_flag wDailyFlags, 4 ; daily unused
```
Comment (Rangi42) [inline data/events/engine_flags.asm:123]:
```suggestion
	engine_flag wDailyFlags, 4 ; unused
```

File: data/events/engine_flags.asm [hunk 11]
```diff
@@ -71,105 +69,104 @@ EngineFlags:
...
-	engine_flag wDailyFlags4, 4 ; changed trendy phrase
-	engine_flag wDailyFlags4, 5 ; all hidden grottoes
+	engine_flag wDailyFlags4, DAILYFLAGS4_SEASHORE_SHELL_BELL_F
+	engine_flag wDailyFlags4, DAILYFLAGS4_SHAMOUTI_RESTAURANT_CHALLENGE_F
+	engine_flag wDailyFlags4, DAILYFLAGS4_FOUGHT_PSYCHIC_INVER_F
+	engine_flag wDailyFlags4, DAILYFLAGS4_BOUGHT_MOOMOO_MILK_F
+	engine_flag wDailyFlags4, DAILYFLAGS4_CHANGED_TRENDY_PHRASE_F
+	engine_flag wDailyFlags4, DAILYFLAGS4_ALL_HIDDEN_GROTTOES_F
 	engine_flag wDailyFlags4, 6 ; daily unused
 	engine_flag wDailyFlags4, 7 ; daily unused
```
Comment (Rangi42) [inline data/events/engine_flags.asm:153]:
```suggestion
	engine_flag wDailyFlags4, 6 ; unused
	engine_flag wDailyFlags4, 7 ; unused
```

File: data/wild/unlocked_unowns.asm [hunk 12]
```diff
@@ -6,17 +6,20 @@ MACRO unown_set
...
-	dw .Set2
-	dw .Set3
-	dw .Set4
+; entries correspond to UNLOCKED_UNOWNS_* constants
+	table_width 2, UnlockedUnownLetterSets
+	dw .Set_A_J
+	dw .Set_K_Q
+	dw .Set_R_W
+	dw .Set_X_Question
+	assert_table_length NUM_UNOWN_SETS
```
Comment (Rangi42) [inline data/wild/unlocked_unowns.asm:15]:
```suggestion
	assert_table_length NUM_UNLOCKED_UNOWN_SETS
```

File: data/wild/unlocked_unowns.asm [hunk 13]
```diff
@@ -6,17 +6,20 @@ MACRO unown_set
...
 .End
```
Comment (Rangi42) [inline data/wild/unlocked_unowns.asm:16]: This can be removed now that `NUM_UNLOCKED_UNOWN_SETS` exists.

File: engine/events/halloffame.asm [hunk 14]
```diff
@@ -13,7 +13,7 @@ HallOfFame::
 
 	; Enable the Pokégear map to cycle through all of Kanto
 	ld hl, wStatusFlags
-	set 6, [hl] ; hall of fame
+	set STATUSFLAGS_HALL_OF_FAME_F, [hl] ; hall of fame
```
Comment (Rangi42) [inline engine/events/halloffame.asm:16]:
```suggestion
	set STATUSFLAGS_HALL_OF_FAME_F, [hl]
```

File: engine/menus/intro_menu.asm [hunk 15]
```diff
@@ -452,7 +452,7 @@ Continue_LoadMenuHeader:
 	ldh [hBGMapMode], a
 	ld hl, .MenuDataHeader_Dex
 	ld a, [wStatusFlags]
-	bit 0, a ; pokedex
+	bit STATUSFLAGS_POKEDEX_F, a ; pokedex
```
Comment (Rangi42) [inline engine/menus/intro_menu.asm:455]:
```suggestion
	bit STATUSFLAGS_POKEDEX_F, a
```

**Comment by Rangi42** [comment]
@vulcandth Thanks for taking this on! It'll need rebasing to fix a new merge conflict, I noticed some optimizations during review.

File: engine/menus/intro_menu.asm [hunk 16]
```diff
@@ -407,8 +407,8 @@ FinishContinueFunction:
 	ld [wDontPlayMapMusicOnReload], a
 	ld [wLinkMode], a
 	ld hl, wGameTimerPaused
-	set 0, [hl]
-	res 7, [hl]
+	set GAME_TIMER_PAUSED_F, [hl]
+	res GAME_TIMER_MOBILE_F , [hl]
```
Comment (vulcandth) [inline engine/menus/intro_menu.asm:411]: Oh, I think this flag can be removed.. I don't think its used anywhere else not that the mobile code has been stripped. May want to verify.

Comment (Rangi42) [reply engine/menus/intro_menu.asm:411]: Yeah, it was only ever `set` in mobile code.

**Pull request merged**